src/python/pants/engine/build_files.py

```diff
@@ -110,7 +109,8 @@ def resolve_unhydrated_struct(address_family, address):
   """
 
   struct = address_family.addressables.get(address)
-  if not struct:
+  addresses = address_family.addressables
+  if not struct or address not in addresses:
```

Comment:
Can the struct value in addressables be a false-ish value? Otherwise I think you don't need the or.

If it can be false-ish, are there any usages below that might fail if it is false-ish?